Add dismiss & reject buttons back to /reporteddefs page

austinratcliff opened the pull request:
I removed `this.isAuthenticated` from the `ResultCard` component since:

- The `isAuthenticated` prop isn't being passed to `<ResultCard />` in `PotentialDefs.js` nor `ReportedDefs.js`.
- `this.isAuthenticated` is being used for only potential defs in `ResultCard.js`, not reported defs.
- Both `PotentialDefs.js` and `ReportedDefs.js` handle authentication already.

As far as I know, this removal doesn't affect other parts of the app and allowed the `Dismiss` and `Reject` buttons to reappear. The functionality for each button is already defined in `ReportedDefs.js`.

saragw6 commented:
I haven't been able to confirm in the UI because I'm getting a weird error, but it seems like this will make these buttons visible on any reported definition to all users, even when not logged in as an admin.

austinratcliff commented:
Summary of new changes —

- I reverted my initial commit to add `this.isAuthenticated` back to the `ResultCard` component since that wasn't the issue
- I passed the `auth` prop to the `ResultList` component since that _was_ the issue
- I added `window.location.pathname === '/reporteddefs'` as an additional condition to render a reported def card's buttons so that they won't render on any other page when logged in (this is a quick solution, so I'm totally open to any other direction that you may want to take for this)
- I merged the changes from PR #59 since they were conflicting with this PR, but I didn't realize that doing so in GitHub would add a new commit, so I'm sorry for that ugly commit 😅
